rp2040-hal/src/multicore.rs

```diff
@@ -1,5 +1,41 @@
 //! Multicore support
-// See [Chapter ?? Section ??](https://datasheets.raspberrypi.org/rp2040/rp2040_datasheet.pdf) for more details
+//!
+//! This module handles setup of the 2nd cpu core on the rp2040, which we refer to as core1.
+//! It provides functionality for setting up the stack, and starting core1.
+//!
+//! The options for an entrypoint for core1 are
+//! - a function that never returns - eg
+//! `fn core1_task() -> ! { loop{} }; `
+//! - a lambda (note: This requires a global allocator which requires a nightly compiler. Not recommended for beginners)
+//!
+//! # Usage
+//!
+//! ```no_run
+//! #[repr(C, align(32))]
+//! pub struct Stack<const SIZE: usize> {
```

Comment:
Maybe `Stack` should be in `multicore` - seeing as pretty much everyone is going to want it.

Reply:
I'll bring it into multicore.rs

Also still worth considering whether we switch multicore from taking a **&mut slice** to taking a **struct Stack** if we expect users to always use the type to avoid pulling out the inner slice, as per @jannic's suggestion:

https://github.com/rp-rs/rp-hal/compare/main...jannic:user-stack-aligned

Reply:
Excellent. Maybe a `::new()` as well? ;)

Reply:
Okay, you twisted my arm :D

Reply:
Just curious: Do you happen to know if the rust compiler reliably inlines `new` and avoids copying the created object?
(I guess in  this case it wouldn't matter much as long as you don't try to give half of your own stack to the other core.)

Reply:
In this case it's a const fn so it's always going to be.
It has to be if it's going to initialise a static object in Rust, since there's no life before main.
In the general case I think it depends on what you're doing in new().